src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,113 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+
+/**
+ * Parser class which handles the processing of the inputs detected
+ * by the User Interface.
+ */
+
+public class Parser {
+    /**
+     * Parses the input from the user.
+     *
+     * @param input The input inserted by the user.
+     * @return The respective commands to be executed based on the input.
+     * @throws DukeException This exception is thrown when the input is invalid.
+     */
+    public static Command parse(String input) throws DukeException {
```

Comment:
This method / while loop might be too long. This [coding quality website](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)  stated the following:

>Be wary when a method is longer than the computer screen, and take corrective action when it goes
>beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.

This is a little longer than 30 LOC , though I too am not sure how to condense this part of logic too haha